apps/electron/src/ipc/ipc-listeners-manager.ts

```diff
@@ -372,5 +373,25 @@ export class IpcListenersManager {
     });
 
 
+  }
+
+  private setupFreecompanyWorkshopsListeners(): void {
+    const freecompanyWorkshopsPath = join(app.getPath('userData'), 'freecompany-workshops.json');
+
+    ipcMain.on('freecompany-workshops:set', (event, inventory) => {
+      writeFileSync(freecompanyWorkshopsPath, JSON.stringify(inventory));
+    });
+
+    ipcMain.on('freecompany-workshops:get', (event, inventory) => {
+      readFile(freecompanyWorkshopsPath, 'utf8', (err, content) => {
+        if (err) {
+          event.sender.send('freecompany-workshops:value', {});
+        } else {
+          event.sender.send('freecompany-workshops:value', JSON.parse(content));
```

Comment:
Just a small guard for the case where the file is empty (yeah, it happens 😩)
```suggestion
          event.sender.send('freecompany-workshops:value', JSON.parse(content || '{}'));
```